File: tests/html/css_test.dart
library CssTest;

import 'package:unittest/unittest.dart';
import 'package:unittest/html_individual_config.dart';
import 'dart:html';

main() {
  useHtmlIndividualConfiguration();

  group('supportsPointConversions', () {
    test('supported', () {
      expect(Window.supportsPointConversions, true);
    });
  });

  group('functional', () {
    test('DomPoint', () {
      var expectation =
          Window.supportsPointConversions ? returnsNormally : throws;
      expect(() {
        Element element = new Element.tag('div');
        element.attributes['style'] = '''
          position: absolute;
          width: 60px;
          height: 100px;
          left: 0px;
          top: 0px;
          background-color: red;
          -webkit-transform: translate3d(250px, 100px, 0px);
          -moz-transform: translate3d(250px, 100px, 0px);
          ''';
        document.body.append(element);

        var elemRect = element.getBoundingClientRect();

        checkPoint(250, 100, new Point(elemRect.left, elemRect.top));
        checkPoint(310, 200, new Point(elemRect.right, elemRect.bottom));
      }, expectation);
    });
  });
}

void checkPoint(expectedX, expectedY, Point point) {
  expect(point.x.round(), equals(expectedX), reason: 'Wrong point.x');
  expect(point.y.round(), equals(expectedY), reason: 'Wrong point.y');
}

File: pkg/dev_compiler/test/codegen_expected/lib/html/css_test.js
dart_library.library('lib/html/css_test', null, /* Imports */[
  'dart_sdk',
  'unittest'
], function load__css_test(exports, dart_sdk, unittest) {
  'use strict';
  const core = dart_sdk.core;
  const html = dart_sdk.html;
  const math = dart_sdk.math;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const html_individual_config = unittest.html_individual_config;
  const unittest$ = unittest.unittest;
  const src__matcher__expect = unittest.src__matcher__expect;
  const src__matcher__core_matchers = unittest.src__matcher__core_matchers;
  const src__matcher__throws_matcher = unittest.src__matcher__throws_matcher;
  const css_test = Object.create(null);
  let PointOfnum = () => (PointOfnum = dart.constFn(math.Point$(core.num)))();
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  let VoidTovoid = () => (VoidTovoid = dart.constFn(dart.definiteFunctionType(dart.void, [])))();
  let dynamicAnddynamicAndPointOfnumTovoid = () => (dynamicAnddynamicAndPointOfnumTovoid = dart.constFn(dart.definiteFunctionType(dart.void, [dart.dynamic, dart.dynamic, PointOfnum()])))();
  css_test.main = function() {
    html_individual_config.useHtmlIndividualConfiguration();
    unittest$.group('supportsPointConversions', dart.fn(() => {
      unittest$.test('supported', dart.fn(() => {
        src__matcher__expect.expect(html.Window[dartx.supportsPointConversions], true);
      }, VoidTodynamic()));
    }, VoidTovoid()));
    unittest$.group('functional', dart.fn(() => {
      unittest$.test('DomPoint', dart.fn(() => {
        let expectation = dart.test(html.Window[dartx.supportsPointConversions]) ? src__matcher__core_matchers.returnsNormally : src__matcher__throws_matcher.throws;
        src__matcher__expect.expect(dart.fn(() => {
          let element = html.Element.tag('div');
          element[dartx.attributes][dartx.set]('style', '          position: absolute;\n          width: 60px;\n          height: 100px;\n          left: 0px;\n          top: 0px;\n          background-color: red;\n          -webkit-transform: translate3d(250px, 100px, 0px);\n          -moz-transform: translate3d(250px, 100px, 0px);\n          ');
          html.document[dartx.body][dartx.append](element);
          let elemRect = element[dartx.getBoundingClientRect]();
          css_test.checkPoint(250, 100, new (PointOfnum())(elemRect[dartx.left], elemRect[dartx.top]));
          css_test.checkPoint(310, 200, new (PointOfnum())(elemRect[dartx.right], elemRect[dartx.bottom]));
        }, VoidTodynamic()), expectation);
      }, VoidTodynamic()));
    }, VoidTovoid()));
  };
  dart.fn(css_test.main, VoidTodynamic());
  css_test.checkPoint = function(expectedX, expectedY, point) {
    src__matcher__expect.expect(point.x[dartx.round](), src__matcher__core_matchers.equals(expectedX), {reason: 'Wrong point.x'});
    src__matcher__expect.expect(point.y[dartx.round](), src__matcher__core_matchers.equals(expectedY), {reason: 'Wrong point.y'});
  };
  dart.fn(css_test.checkPoint, dynamicAnddynamicAndPointOfnumTovoid());
  // Exports:
  exports.css_test = css_test;
});
